runsc/specutils/nvidia.go

```diff
@@ -23,16 +23,35 @@ import (
 	"strings"
 
 	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"gvisor.dev/gvisor/pkg/log"
 	"gvisor.dev/gvisor/runsc/config"
 )
 
 const nvdEnvVar = "NVIDIA_VISIBLE_DEVICES"
 
+// annotationNVProxy enables nvproxy.
+const annotationNVProxy = "dev.gvisor.spec.nvproxy"
+
+// NVProxyEnabled checks both the nvproxy annotation and conf.NVProxy to see if nvproxy is enabled.
+func NVProxyEnabled(spec *specs.Spec, conf *config.Config) bool {
+	for k, v := range spec.Annotations {
+		if k != annotationNVProxy {
+			continue
+		}
+		if v == "true" {
+			return true
+		}
+		log.Warningf("nvproxy annotation set to invalid value %q. Skipping.", v)
+		break
+	}
```

Comment:
Since `spec.Annotations` is a map, it might be better to directly index the value using the key:

```go
	if v, ok := spec.Annotations[annotationNVProxy]; ok {
		if v == "true" {
			return true
		}   
		log.Warningf("nvproxy annotation set to invalid value %q. Skipping.", v)
	}
```

Reply:
Thanks for the review :)